### PR TITLE
Fixes #276: Allow passing a sitegroup to all IDE commands.

### LINKED ISSUE #276: Allow passing a sitegroup to all IDE commands
Doing this would prevent being prompted to select an application first always, and then getting the information we need only. This would also reduce the number of web requests to the Cloud Platform API.

Here's how it looks today: 

```
$ acli ide:list
Please select a Cloud Platform application:
  [0] Corporate Site
  [1] Commerce portal
 > 0
Would you like to link the Cloud application Corporate Site to this repository? n
 ======================================================================= 
  IDEs                                                                   
 ======================================================================= 
  corpsite (foo@bar.baz)                                  
  IDE URL: https://d6l3c150-7ad7-49ff-be21-4ed45767760a.ide.ahdev.cloud  
  Web URL: https://d6l3c150-7ad7-49ff-be21-4ed45767760a.web.ahdev.cloud  
 ======================================================================= 
  portal (foo@bar.baz)                                     
  IDE URL: https://3a4a5c8c-2732-4e5c-adeb-eb3082169k91.ide.ahdev.cloud  
  Web URL: https://3a4a5c8c-2732-4e5c-adeb-eb3082169k91.web.ahdev.cloud  
 ======================================================================= 
```

Here's how it could look tomorrow:

```
$ acli ide:list portald9
Please select a Cloud Platform application:
  [0] Corporate Site
  [1] Commerce portal
 > 1
Would you like to link the Cloud application Corporate Site to this repository? n
 ======================================================================= 
  IDEs                                                                   
 ======================================================================= 
  portal (foo@bar.baz)                                     
  IDE URL: https://3a4a5c8c-2732-4e5c-adeb-eb3082169k91.ide.ahdev.cloud  
  Web URL: https://3a4a5c8c-2732-4e5c-adeb-eb3082169k91.web.ahdev.cloud  
 ======================================================================= 
```

This would apply to all commands below:

* `ide:create <sitegroup>`
* `ide:delete <sitegroup>`
* `ide:list <sitegroup>`
* `ide:open <sitegroup>`

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

File: src/Command/Ide/IdeCreateCommand.php [hunk 1]
```diff
@@ -38,8 +39,7 @@ class IdeCreateCommand extends CommandBase {
    */
   protected function configure() {
     $this->setDescription('Create a Cloud IDE for development')
-    ->addOption('cloud-app-uuid', 'uuid', InputOption::VALUE_REQUIRED, 'The UUID of the associated Cloud Platform Application');
```
Comment (grasmash) [inline src/Command/Ide/IdeCreateCommand.php:41]: Unfortunately I think this technically breaks backwards compatibility.

File: src/Command/Ide/IdeDeleteCommand.php [hunk 2]
```diff
@@ -22,7 +23,7 @@ class IdeDeleteCommand extends IdeCommandBase {
    */
   protected function configure() {
     $this->setDescription('Delete a Cloud IDE')
-    ->addOption('cloud-app-uuid', 'uuid', InputOption::VALUE_REQUIRED, 'The UUID of the associated Cloud Platform Application');
+      ->addArgument('applicationUuid', InputArgument::OPTIONAL, 'The UUID or alias of the associated Cloud Platform Application');
```
Comment (grasmash) [inline src/Command/Ide/IdeDeleteCommand.php:26]: I wonder if we should instead have an optional argument for the Cloud IDE UUID here.

File: src/Command/Ide/IdeOpenCommand.php [hunk 3]
```diff
@@ -20,8 +21,8 @@ class IdeOpenCommand extends IdeCommandBase {
    */
   protected function configure() {
     $this->setDescription('Open a Cloud IDE in your browser')
-    ->addOption('cloud-app-uuid', 'uuid', InputOption::VALUE_REQUIRED, 'The UUID of the associated Cloud Platform Application')
-    ->setHidden(AcquiaDrupalEnvironmentDetector::isAhIdeEnv());
+      ->addArgument('applicationUuid', InputArgument::OPTIONAL, 'The UUID or alias of the associated Cloud Platform Application')
```
Comment (grasmash) [inline src/Command/Ide/IdeOpenCommand.php:24]: I wonder if we should instead have an optional argument for the Cloud IDE UUID here.

**Review by danepowell — APPROVED** [review]
Nice, this is exactly how I was thinking we should handle it... allow passing a sitegroup / appid for any command, not just IDEs.

If we have to break BC, it's worth it, we'll want to make sure to call it out in the release notes.

On BLT, to make sure things like this don't get forgotten in release notes, we have a "change record" label for Github issues like this, as well as a step in the release runbook to check for these when writing release notes. Is that helpful here, or overkill?

**Comment by grasmash** [comment]
No I don’t think it’s overkill.

**Pull request merged**